app/models/manageiq/providers/kubevirt/builder.rb

```diff
@@ -22,10 +22,32 @@
 class ManageIQ::Providers::Kubevirt::Builder
   class << self
     def build_inventory(manager, target)
+      persister = ManageIQ::Providers::Kubevirt::Inventory::Persister.new(manager, target)
+
       # Create the collector, parser and persister:
-      collector = ManageIQ::Providers::Kubevirt::Inventory::Collector.new(manager, target)
-      parser = ManageIQ::Providers::Kubevirt::Inventory::Parser.new
-      persister = ManageIQ::Providers::Kubevirt::Inventory::Persister.new(manager, target, collector)
+      if target.kind_of?(ManageIQ::Providers::Kubevirt::InfraManager::Vm)
+        name = target.name
+        collector = ManageIQ::Providers::Kubevirt::Inventory::Collector.new(manager, target)
+        collector.nodes = {}
+        manager.with_provider_connection do |connection|
+          collector.offline_vms = connection.offline_vm(name)
+          collector.live_vms = connection.live_vm(name)
+          collector.templates = connection.template(name)
+        end
+
+        parser = ManageIQ::Providers::Kubevirt::Inventory::Parser::PartialRefresh.new
+      else
+        collector = ManageIQ::Providers::Kubevirt::Inventory::Collector.new(manager, manager)
```

Comment:
this looks very similar to https://github.com/ManageIQ/manageiq-providers-kubevirt/blob/master/app/models/manageiq/providers/kubevirt/infra_manager/refresh_worker/runner.rb#L118

Can we reuse this builder for creating the full refresh collector, parser and persister ?
This will allow us also to cover the same code with the exact spec file.

Reply:
Good idea, will do.